### PR TITLE
MAT-8038: add new endpoint for library expansions

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-8038](https://jira.cms.gov/browse/MAT-8038)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java [hunk 1]
```diff
@@ -119,6 +120,21 @@ public ResponseEntity<LibrarySetDTO> getLibrarySetBySetId(@PathVariable String s
     return ResponseEntity.ok(cqlLibraryService.getLibrarySetBySetId(setId));
   }
 
+  @GetMapping("/byLibrarySetId")
+  public ResponseEntity<List<LibraryListDTO>> getLibrariesByLibrarySetId(
+      @RequestParam(name = "librarySetId") String librarySetId, boolean sortByLatestVersion) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java]: sortByLatestVersion is not annotated with RequestParam, Is there a reason ?

File: src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibraryRepository.java [hunk 2]
```diff
@@ -39,4 +40,9 @@ List<CqlLibrary> findAllByCqlLibraryNameAndDraftAndVersionAndModel(
 
   List<CqlLibrary> findByLibrarySetIdAndDraftAndActive(
       String librarySetId, boolean draft, boolean active);
+
+  List<LibraryListDTO> findLibrariesByLibrarySetIdAndActiveIsTrueOrderByVersionDesc(
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibraryRepository.java]: Will be nice to have dynamic value for Active and sortByVersion numnber

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java [hunk 3]
```diff
@@ -119,6 +120,21 @@ public ResponseEntity<LibrarySetDTO> getLibrarySetBySetId(@PathVariable String s
...
+    List<LibraryListDTO> cqlLibraries =
+        cqlLibraryService.getLibrariesByLibrarySetId(librarySetId, sortByLatestVersion);
+    cqlLibraries.forEach(
+        library -> {
+          LibrarySet librarySet =
+              librarySetRepository.findByLibrarySetId(library.getLibrarySetId()).orElse(null);
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:132]: We recently had this discussion of not perfoming individual calls to DB to fetch those librarySets, when data grew this operation could slow down the response time.

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:132]: Either by doing a bulk operation or should include LibrarySet in the intial call by combining both collections together

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java (hunk 1 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java]: no real reason, but since we are making sort dynamic I added it in with a default of true

File: src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibraryRepository.java (hunk 2 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibraryRepository.java]: Added for sort, but as MADIE doesn't really want to show inactive data I left active hardcoded to true. I did make a slight change to make it easier to update for a more dynamic endpoint if that is a direction that MADIE goes in the future.

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java (hunk 3 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java:132]: The DB calls that we talked about that was dropping performance in measures was a different call. That being said this isn't good for performance either. But is caused by a much larger issue being how the database and the data models are configured that don't allow us to make those necessary joins when retrieving data. MAT-8485 will have to be used in order to investigate these performance issues, so that we know the best way to move forward. Until then, we unfortunately need to do this.

**Review by RohitKandimalla — APPROVED** [review]
